packages/editor/src/editor/sidebar/fields/InputFieldWithBrowser.tsx

```diff
@@ -19,27 +21,39 @@ export const InputFieldWithBrowser = ({
 }: InputFieldProps & { browsers: Array<BrowserType>; options?: TextBrowserFieldOptions }) => {
   const [open, setOpen] = useState(false);
   const inputRef = useRef<HTMLInputElement>(null);
+  const cmsQuickFixPopoverRef = useRef<HTMLDivElement>(null);
+  const { isFocusWithin, focusWithinProps } = useOnFocus(value, onChange);
   const { handleTextSelection, showQuickFix, getSelectedText, selection } = useTextSelection(inputRef);
 
   return (
     <Dialog open={open} onOpenChange={setOpen}>
-      <BasicField label={label} message={message} style={{ flex: '1' }}>
-        <InputGroup>
-          <Input
-            ref={inputRef}
-            value={value}
-            onChange={e => onChange(e.target.value)}
-            onSelect={() => handleTextSelection()}
-            onBlur={onBlur}
-            placeholder={options?.placeholder}
-          />
-          {showQuickFix() && selection && browsers.some(b => b === 'CMS') && (
-            <AddCmsQuickFixPopover value={getSelectedText()} selection={selection} inputRef={inputRef} onChange={onChange} />
-          )}
-          <DialogTrigger asChild>
-            <Button icon={IvyIcons.ListSearch} aria-label='Browser' />
-          </DialogTrigger>
-        </InputGroup>
+      <BasicField label={label} message={message} style={{ flex: '1' }} {...focusWithinProps} className='badge-field' tabIndex={0}>
+        {isFocusWithin || cmsQuickFixPopoverRef.current ? (
```

Comment:
why do you need the ref here?
I think you can also use just the open state or not?

Reply:
this is used to determine if the AddCmsQuickFixPopover is open or not, the open state is used for the dialog, not the AddCmsQuickFixPopover

Reply:
but isn't the browser still open if the popover is open?

Reply:
The AddCmsQuickFixPopover does not use the browser, so it isn't open.

Reply:
ah right.